src/Phobos.Kafka/src/Petabridge.Phobos.Kafka.Producer/PhobosActorRefSourceActor.cs

```diff
@@ -0,0 +1,205 @@
+        /// <summary>
+        /// TBD
+        /// </summary>
+        public readonly OverflowStrategy OverflowStrategy;
+        private ILoggingAdapter _log;
+
+        /// <summary>
+        /// TBD
+        /// </summary>
+        /// <param name="bufferSize">TBD</param>
+        /// <param name="overflowStrategy">TBD</param>
+        /// <param name="maxFixedBufferSize">TBD</param>
+        public PhobosActorRefSourceActor(int bufferSize, OverflowStrategy overflowStrategy, int maxFixedBufferSize)
+        {
+            BufferSize = bufferSize;
+            OverflowStrategy = overflowStrategy;
+            Buffer = bufferSize != 0 ?  Producer.Buffer.Create<(T, ISpanContext)>(bufferSize, maxFixedBufferSize) : null;
+        }
+
+        /// <summary>
+        /// TBD
+        /// </summary>
+        protected ILoggingAdapter Log => _log ??= Context.GetLogger();
+
+        /// <summary>
+        /// TBD
+        /// </summary>
+        /// <param name="message">TBD</param>
+        /// <returns>TBD</returns>
+        protected override bool Receive(object message)
+            => DefaultReceive(message) || RequestElement(message) || (message is T && ReceiveElement((T) message));
+
+        /// <summary>
+        /// TBD
+        /// </summary>
+        /// <param name="message">TBD</param>
+        /// <returns>TBD</returns>
+        protected bool DefaultReceive(object message)
+        {
+            if (message is Cancel)
+                Context.Stop(Self);
+            else if (message is Status.Success)
+            {
+                if (BufferSize == 0 || Buffer.IsEmpty)
+                    Context.Stop(Self);  // will complete the stream successfully
+                else
+                    Context.Become(DrainBufferThenComplete);
+            }
+            else if (message is Status.Failure && IsActive)
+                OnErrorThenStop(((Status.Failure)message).Cause);
+            else
+                return false;
+            return true;
+        }
+
+        /// <summary>
+        /// TBD
+        /// </summary>
+        /// <param name="message">TBD</param>
+        /// <returns>TBD</returns>
+        protected virtual bool RequestElement(object message)
+        {
+            if (message is Request)
+            {
+                // totalDemand is tracked by base
+                if (BufferSize != 0)
+                    while (TotalDemand > 0L && !Buffer.IsEmpty)
+                        OnNext(Buffer.Dequeue());
+
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// TBD
+        /// </summary>
+        /// <param name="element">TBD</param>
+        /// <returns>TBD</returns>
+        protected virtual bool ReceiveElement(T element)
+        {
+            if (IsActive)
+            {
+                var context = Context.GetInstrumentation().ActiveSpan?.Context;
+                var message = (element, context);
```

Comment:
This pushes the span context together with the message as a tuple into the stream.
Is there an easier way to do this?